### PR TITLE
CCS-3899 keycloak oidc filter

### PR DESCRIPTION
**Pull request by xdavidson**
This PR integrates with external SSO using KeycloakOIDCFitler 

-  if SSO is enabled, user will be prompted to the external sso for authentication. Once authenticated, if user is new, it creates the user locally and grants it "pantheon-author" role
- If SSO is not enabled, aka. AUTH_SERVER_URL is not defined as an env variable, it falls back to the basic auth. 
-  Introduce a pantheon-administrator groups that allows its member to access content browser and Felix console
-  The configuration for build/deploy pipeline is a separate PR, which is going to be handled in gitlab.

### CODE REVIEW

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/user/AuthUserInfoServlet.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ *   userId: default value "anonymous"
+ *   groups: default value "[]"
+ *   userType: default value ""
+ *
+ * @author Lisa Davidson
+ */
+@Component(
+        service = Servlet.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Servlet to fetch the current build date in the Admin Panel",
```
Comment (carlosmunoz) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/auth/user/AuthUserInfoServlet.java:39]: minor: update the service description for this servlet

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/user/AuthUserInfoServlet.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+            if (currentUserGroups != null) {
+                while (currentUserGroups.hasNext()) {
+                    Group grp = (Group) currentUserGroups.next();
+                    String groupName = grp.getID();
+                    ((List) userInfo.get("groups")).add(groupName);
+                }
+            }
+
+            } catch (Exception e) {
```
Comment (carlosmunoz) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/auth/user/AuthUserInfoServlet.java:67]: minor: code indentation

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/user/AuthUserInfoServlet.java (hunk 1 above)
Comment (xdavidson) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/auth/user/AuthUserInfoServlet.java]: done

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/user/AuthUserInfoServlet.java (hunk 2 above)
Comment (xdavidson) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/auth/user/AuthUserInfoServlet.java]: done

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/keycloak/KeycloakAuthenticationHandler.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                            session.logout();
+                            resolver.commit();
+                        }
+                        Session session = this.repository.login(new SimpleCredentials(extractedUserId, extractedUserId.toCharArray()));
+                            if (session != null) {
+                            return new AuthenticationInfo(AUTH_TYPE, session.getUserID(), session.getUserID().toCharArray());
+                        }
+                    }
+                } catch (Exception e) {
+                    log.warning("[" + KeycloakAuthenticationHandler.class.getSimpleName() + "] Exception in extractCredentials while processing the request" + e.getMessage());
```
Comment (aprajshekhar) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/auth/keycloak/KeycloakAuthenticationHandler.java:100]: Wouldn't log.error be better?

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/keycloak/KeycloakFilter.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        }
+        // Load client configuration information
+        File file = new File( System.getProperty( KARAF_ETC ) + File.separator + KEYCLOAKOIDCFILTER_CONFIG_FILE_NAME );
+
+        // load config from the file system
+        InputStream is = null;
+        try {
+            is = new FileInputStream(file);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
```
Comment (aprajshekhar) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/auth/keycloak/KeycloakFilter.java:82]: Can we use logger?

**Comment by aprajshekhar** [comment]
Great work! Just two minor comments.

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/keycloak/KeycloakAuthenticationHandler.java (hunk 3 above)
Comment (xdavidson) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/auth/keycloak/KeycloakAuthenticationHandler.java]: updated

File: pantheon-bundle/src/main/java/com/redhat/pantheon/auth/keycloak/KeycloakFilter.java (hunk 4 above)
Comment (xdavidson) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/auth/keycloak/KeycloakFilter.java]: updated

**Review by aprajshekhar — APPROVED** [review]
LGTM.

**Pull request merged**